subprojects/core/src/main/java/org/gradle/execution/CompositeAwareTaskSelector.java

```diff
@@ -104,6 +108,21 @@ private BuildState findIncludedBuild(Path taskPath) {
         return null;
     }
 
+    private BuildState findIncludedBuild(File root) {
+        if (root == null || buildStateRegistry.getIncludedBuilds().isEmpty()) {
```

Comment:
I'd do without the isEmpty check here - the effect is the same and I think it would be easier for a reader.

Reply:
Fixed.